SDWebImage/Core/SDAnimatedImageView.m

```diff
@@ -183,8 +183,11 @@ - (void)setImage:(UIImage *)image
         
         // Ensure disabled highlighting; it's not supported (see `-setHighlighted:`).
         super.highlighted = NO;
-        
+    #if SD_UIKIT
         [self stopAnimating];
+    #else
```

Comment:
This code don't need. Because SDAnimatedImageView provide the `startAnimating` and `stopAnimating` for macOS.

The only thing not exist is the `isAnimting` property.

You can actually check the code again

Reply:
Do you mean to call [self stopAnimating] directly here without distinguishing between iOS and Mac?
However, if we do not call [self setAnimates: NO] in the Mac environment, the animates property will not change.

Reply:
> However, if we do not call [self setAnimates: NO] in the Mac environment, the animates property will not change.

Previous code does not call that. I just review the code based on changes diff.

If you think this is a bug, maybe we can have a fix at the same time :)

Reply:
like:

```
- (void)startAnimating
{
    if (self.player) {
        [self updateShouldAnimate];
        if (self.shouldAnimate) {
            [self.player startPlaying];
        }
    } else {
#if SD_UIKIT
        [super startAnimating];
#else
        [super setAnimates:YES];
#endif
    }
}
```

Reply:
I did some tweaking, and I overwrote the - (BOOL)animates function